cmd/goreleaser/internal/configure.go

```diff
@@ -67,26 +72,42 @@ func Builds(dist string) []config.Build {
 // Build configures a goreleaser build.
 // https://goreleaser.com/customization/build/
 func Build(dist string) config.Build {
+	var goos []string
+	var archs []string
+	var ignore []config.IgnoredBuild
+	var armVersions []string
+	flags := []string{"-trimpath"}
+	if dist == "otelcol-k8s" {
+		goos = K8sGoos
+		archs = K8sArchs
+		ignore = make([]config.IgnoredBuild, 0)
+		armVersions = make([]string, 0)
+	} else {
+		goos = []string{"darwin", "linux", "windows"}
+		archs = Architectures
+		ignore = []config.IgnoredBuild{
+			{Goos: "darwin", Goarch: "386"},
+			{Goos: "darwin", Goarch: "arm"},
+			{Goos: "darwin", Goarch: "s390x"},
+			{Goos: "windows", Goarch: "arm"},
+			{Goos: "windows", Goarch: "arm64"},
+			{Goos: "windows", Goarch: "s390x"},
+		}
+		armVersions = ArmVersions
+	}
 	return config.Build{
 		ID:     dist,
 		Dir:    "_build",
 		Binary: dist,
 		BuildDetails: config.BuildDetails{
 			Env:     []string{"CGO_ENABLED=0"},
-			Flags:   []string{"-trimpath"},
+			Flags:   flags,
```

Comment:
It doesn't look like the `flags` variable is used elsewhere yet so this can stay as `[]string{"-trimpath"},`